example/helloWorld/src/helloworld.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * \file
+ * Copyright 2014-2015 Benjamin Worpitz
+ *
+ * This file is part of alpaka.
+ *
+ * alpaka is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * alpaka is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with alpaka.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * \author Erik Zenker
+ *
+ */
+// Alpaka
+#include <alpaka/alpaka.hpp>
+
+// STL
+#include <iostream>
+
+
+/**
+ * Helper functions that calculates the global thread id depending
+ * on the number of threads set by the workdiv.
+ */
+template <typename T_Acc>
+size_t globalThreadIdx(T_Acc const &acc){
+    auto threadsExtent   = alpaka::workdiv::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+    auto threadIdx       = alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+
+    /**
+     * Map the three dimensional thread index into a
+     * one dimensional thread index space.
+     */
+    auto globalThreadIdx = alpaka::core::mapIdx<1u>( threadIdx,
+						     threadsExtent );
+    return globalThreadIdx[0];
+}
+
+/**
+ * Hello World Kernel
+ *
+ * Prints "[tid] Hello World" where tid is
+ * the global thread number.
+ */
+struct HelloWorldKernel {
+    template <typename T_Acc>
+    ALPAKA_FN_ACC void operator()( T_Acc const &acc) const {
+        printf("[%u] Hello World\n", static_cast<unsigned>(globalThreadIdx(acc)));
```

Comment:
Could you please log the 3 dimensional as well as the linearized thread index? Because else there is little sense in calling the kernel with 3 dimensions. This would also show that alpaka uses the mathematical index order [z][y][x] where the last index is the fast one (you are calling the kernel with 128 blocks/threads in the z dimension).